group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorMetrics.java

```diff
@@ -319,40 +319,40 @@ public Long value() {
         registry.newGauge(NUM_GENERIC_GROUPS_PREPARING_REBALANCE, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numGenericGroups(GenericGroupState.PREPARING_REBALANCE);
+                return numGenericGroups(ClassicGroupState.PREPARING_REBALANCE);
             }
         });
 
         registry.newGauge(NUM_GENERIC_GROUPS_COMPLETING_REBALANCE, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numGenericGroups(GenericGroupState.COMPLETING_REBALANCE);
+                return numGenericGroups(ClassicGroupState.COMPLETING_REBALANCE);
             }
         });
 
         registry.newGauge(NUM_GENERIC_GROUPS_STABLE, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numGenericGroups(GenericGroupState.STABLE);
+                return numGenericGroups(ClassicGroupState.STABLE);
             }
         });
 
         registry.newGauge(NUM_GENERIC_GROUPS_DEAD, new com.yammer.metrics.core.Gauge<Long>() {
             @Override
             public Long value() {
-                return numGenericGroups(GenericGroupState.DEAD);
+                return numGenericGroups(ClassicGroupState.DEAD);
             }
         });
 
         registry.newGauge(NUM_GENERIC_GROUPS_EMPTY, new com.yammer.metrics.core.Gauge<Long>() {
```

Comment:
Are we keeping the metrics name with "generic" keyword though? I think we should if these metrics have already been released in previous versions, if not, might want to change the metric names to "classic" as well.

Reply:
We don't have any existing metrics using "generic". They are all new in 3.7.

Reply:
Regarding `GENERIC_GROUP_REBALANCES_SENSOR_NAME`, the patch only changes the name of the sensor. The metrics are not touched.